core/async_cancel.rs

```diff
@@ -334,7 +334,7 @@ mod internal {
 
   impl PartialEq for Node {
     fn eq(&self, other: &Self) -> bool {
-      self as *const _ == other as *const _
+      std::ptr::eq(self, other)
```

Comment:
@piscisaureus please take a look, clippy was complaining about previous version and suggested this instead

Reply:
Looks good to me.